formats/src/main/scala/overflowdb/formats/neo4jcsv/Neo4jCsvImporter.scala

```diff
@@ -0,0 +1,244 @@
+        case ColumnDef(Some(name), valueType, true) =>
+          parseArrayProperty(entry, name, valueType).foreach(properties.addOne)
+        case other =>
+          throw new MatchError(s"unhandled case $other")
+      }
+    }
+    assert(id != null, s"no ID column found in line $lineNo")
+    assert(label != null, s"no LABEL column found in line $lineNo")
+
+    val ret = ParsedNodeRowData(id, label, properties.result())
+    logger.debug("parsed line {}: {}", lineNo, ret)
+    ret
+  }
+
+  private def parseEdgeRowData(columnsRaw: Seq[String], lineNo: Int, columnDefs: Map[Int, ColumnDef]): ParsedEdgeRowData = {
+    var startId: Integer = null
+    var endId: Integer = null
+    var label: String = null
+    val properties = Seq.newBuilder[ParsedProperty]
+    columnsRaw.zipWithIndex.foreach { case (entry, idx) =>
+      assert(columnDefs.contains(idx), s"column with index=$idx not found in column definitions derived from headerFile")
+      columnDefs(idx) match {
+        case ColumnDef(_, ColumnType.StartId, _) =>
+          startId = entry.toInt
+        case ColumnDef(_, ColumnType.EndId, _) =>
+          endId = entry.toInt
+        case ColumnDef(_, ColumnType.Type, _) =>
+          label = entry
+        case ColumnDef(Some(name), valueType, false) =>
+          parseProperty(entry, name, valueType).foreach(properties.addOne)
+        case ColumnDef(Some(name), valueType, true) =>
+          parseArrayProperty(entry, name, valueType).foreach(properties.addOne)
+        case other =>
+          throw new MatchError(s"unhandled case $other")
+      }
+    }
+    assert(startId != null, s"no START_ID column found in line $lineNo")
+    assert(endId != null, s"no END_ID column found in line $lineNo")
+    assert(label != null, s"no LABEL column found in line $lineNo")
+
+    val ret = ParsedEdgeRowData(startId, endId, label, properties.result())
+    logger.debug("parsed line {}: {}", lineNo, ret)
+    ret
+  }
+
+  private def parseProperty(rawValue: String, name: String, valueType: ColumnType.Value): Option[ParsedProperty] = {
+    if (rawValue != "" || valueType == ColumnType.String)
+      Some(ParsedProperty(name, parsePropertyValue(rawValue, valueType)))
+    else
+      None
+  }
+
+  private def parseArrayProperty(rawValue: String, name: String, valueType: ColumnType.Value): Option[ParsedProperty] = {
+    val values = rawValue.split(';') // from the docs: "By default, array values are separated by ;"
+    if (values.nonEmpty && values.head != "") { // csv parser always adds one empty string entry...
+      val parsedValues = values.map(parsePropertyValue(_, valueType))
+      Some(ParsedProperty(name, parsedValues))
+    } else {
+      None
+    }
+  }
+
+  private def parsePropertyValue(rawString: String, valueType: ColumnType.Value): Any = {
+    valueType match {
+      case ColumnType.Int => rawString.toInt
+      case ColumnType.Long => rawString.toLong
```

Comment:
In all those `toX` the same as https://github.com/ShiftLeftSecurity/overflowdb/pull/299/files#r863641990 applies